### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by damaki** [annexi-strayline/AURA#1]
The documentation suggests that AURA CLI currently builds on UNIX platforms. And indeed, trying to build on Windows fails due to some headers being included, e.g. `<sys/ioctl.h>` which aren't available on my platform (Windows with GNAT CE 2021).

Are there plans to support Windows in the near future?

**Comment by Richard-Wai** [comment]
It is definitely something we'd like to see added. In theory it shouldn't be too difficult, though we don't often work with GNAT in Windows.

**Comment by thindil** [comment]
@damaki it could be a temporary solution, but, how do you think about using [MSYS2](https://www.msys2.org) and GNAT from there? If I remember correctly, there is a working FSF version of GNAT.  It **should** work for now until the **real** Windows appear.

**Comment by Richard-Wai** [comment]
@thindil @damaki Indeed, if you can satisfy two conditions: 1. Have a basic POSIX environment with fork() and execve() in the C library, and 2. Have GNAT in the path, then AURA should work as built. The only real OS-dependencies it has directly is fork() and execve(). Everything else uses the Ada standard library.

Of course I have not tested that myself, but in theory it seems possible to use something like cygwin or MSYS2.

**Comment by Richard-Wai** [comment]
@damaki  - To update this, I put a pretty serious effort into evaluating the path forward for this, and even started the work, though its priority remains low. However it should be possible to "port" AURA to windows 100% native without cygwin, just mingw-w64. The main pieces of work are dealing with process spawning and monitoring, but I've already confirmed this has a viable path forward. Besides that there are some minor tweaks to the CLI submodule that would also be needed.
